runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -98,9 +120,9 @@ public void seek(long globalAddress) {
     @Override
     public Stream<SMREntry> streamUpTo(long maxGlobal) {
         return streamView.streamUpTo(maxGlobal)
-                .filter(m -> m.getType() == DataType.DATA)
-                .filter(m -> m.getPayload(runtime) instanceof ISMRConsumable)
-                .map(logData -> ((ISMRConsumable)logData.getPayload(runtime)).getSMRUpdates(streamView.getID()))
+                .filter(m -> m.getType() == DataType.DATA || m.getType() == DataType.CHECKPOINT)
```

Comment:
Scott, you have multiple ifs like this (dataType==DATA || DataTppe==CHECKPOINT). Better add in org.corfudb.protocols.wireprotocol.LogData similar attribute like "metadataAware",  for example "contentAware" or the oposite flag "empty" and replace all ifs like this with if (data.isContentArare()) or if (!data.isEmpty()).   Also inspect once again all ifs where you have if (dataType==DATA) too see do you need to do the replace only in the 3-4 places from your review, or to more places.